bindgen/clang.rs

```diff
@@ -1882,6 +1899,64 @@ impl Drop for TranslationUnit {
     }
 }
 
+/// Translation unit used for macro fallback parsing
+pub(crate) struct FallbackTranslationUnit {
+    file_path: String,
+    idx: Box<Index>,
+    tu: TranslationUnit,
+}
+
+impl fmt::Debug for FallbackTranslationUnit {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "FallbackTranslationUnit {{ }}")
+    }
+}
+
+impl FallbackTranslationUnit {
+    /// Create a new fallback translation unit
+    pub(crate) fn new(file: &str, c_args: &[Box<str>]) -> Option<Self> {
+        let f_index = Box::new(Index::new(true, false));
+        let f_translation_unit = TranslationUnit::parse(
+            &f_index,
+            file,
+            c_args,
+            &[],
+            CXTranslationUnit_None,
+        )?;
+        Some(FallbackTranslationUnit {
+            file_path: file.to_owned(),
+            tu: f_translation_unit,
+            idx: f_index,
+        })
+    }
+
+    /// Get reference to underlying translation unit.
+    pub(crate) fn translation_unit(&self) -> &TranslationUnit {
+        &self.tu
+    }
+
+    /// Reparse a translation unit.
+    pub(crate) fn reparse(&mut self, unsaved: &[UnsavedFile]) -> bool {
+        let mut c_unsaved: Vec<CXUnsavedFile> =
+            unsaved.iter().map(|f| f.x).collect();
+        let ret = unsafe {
+            clang_reparseTranslationUnit(
+                self.tu.x,
+                unsaved.len() as c_uint,
+                c_unsaved.as_mut_ptr(),
+                CXSaveTranslationUnit_None,
+            )
+        };
+        ret == 0
+    }
+}
+
+impl Drop for FallbackTranslationUnit {
+    fn drop(&mut self) {
+        let _ = std::fs::remove_file(&self.file_path);
```

Comment:
I am not sure how `bindgen` handles this sort of situation, but I wonder if this should be warned about or similar.

Reply:
Small question, why is this deleting a file? 

Regarding the warning infrastructure. There is some experimental support for it but it is behind the `experimental` feature.

Reply:
This is just to clean up the file we created. I'm happy to 1. leave it there or 2. move the file creation into the `new()` definition to make it clearer that it is a "managed resource".

Reply:
Ah! This I missed, disregard my previous comments about it.

Reply:
Perhaps it would be better to put the file creation and file deletion in impls in the same data structure to avoid confusion. It seems like this was not intuitive to either of you so I'd rather make sure it's clear to anyone modifying the code.